src/citrine/informatics/workflows/design_workflow.py

```diff
@@ -17,21 +17,23 @@ class DesignWorkflow(Resource['DesignWorkflow'], Workflow, AIResourceMetadata):
     ----------
     name: str
         the name of the workflow
-    design_space_id: UUID
+    design_space_id: Optional[UUID]
         the UUID corresponding to the design space to use
     processor_id: Optional[UUID]
         the UUID corresponding to the processor to use
         if none is provided, one matching your design space will be automatically generated
-    predictor_id: UUID
+    predictor_id: Optional[UUID]
         the UUID corresponding to the predictor to use
     description: Optional[str]
         a description of the workflow
 
     """
 
-    design_space_id = properties.UUID('design_space_id')
+    design_space_id = properties.Optional(properties.UUID, 'design_space_id')
     processor_id = properties.Optional(properties.UUID, 'processor_id')
-    predictor_id = properties.UUID('predictor_id')
+    predictor_id = properties.Optional(properties.UUID, 'predictor_id')
+    branch_id: Optional[UUID] = properties.Optional(properties.UUID, 'branch_id')
```

Comment:
partial design workflows also contain `data_source_id`.  We'll probably want to expose this here too

Reply:
That I would not expose, it’s going to confuse everyone and it’s not used downstream at all. @jspeerless do you agree?